gulpfile.ts

```diff
@@ -36,52 +45,108 @@ async function fetchContentsByPath(repoPath = 'models'): Promise<GithubObject[]>
 }
 
 function generateAPIModel(model: GithubObject): APIModel {
-  const outputFolder = path.basename(path.dirname(model.path))
-  const out = `src/api-models/${outputFolder}`
-  const command = `openapi-generator-cli generate -g typescript-axios --additional-properties=supportsES6=true,useSingleRequestParameter=true --type-mappings=set=Array --skip-validate-spec -o ${out} -i ${model.download_url}`
+  const modelName = path.basename(path.dirname(model.path))
+  const outputPath = `src/api-models/${modelName}`
+  const command = `openapi-generator-cli generate -g typescript-axios --additional-properties=supportsES6=true,useSingleRequestParameter=true --type-mappings=set=Array --skip-validate-spec -o ${outputPath} -i ${model.download_url}`
 
   return {
     ...model,
     command,
+    outputPath,
+    modelName,
   }
 }
 
-function executeGeneratorCLI(model: APIModel): APIModel {
-  // TODO: generate model from API model
-
-  return model
+async function executeGeneratorCLI(model: APIModel): Promise<APIModel> {
+  return new Promise((resolve, reject) => {
+    exec(model.command, (error: ExecException | null) => {
+      if (error) reject(error)
+      else resolve(model)
+    })
+  })
 }
 
-function removeRedundantObjects(model: APIModel): APIModel {
+async function removeRedundantObjects(model: APIModel): Promise<APIModel> {
   /**
-   * TODO: clean up:
+   * Clean up:
    *- .openapi-generator
    *- .gitignore
    *- .openapi-generator-ignore
    *- git_push.sh
    */
 
-  return model
+  return Promise.all([
+    ...redundantFiles.map((object) => fsPromises.unlink(`${model.outputPath}/${object}`)),
+    ...redundantDirectories.map((object) =>
+      fsPromises.rmdir(`${model.outputPath}/${object}`, { recursive: true }),
+    ),
+  ]).then((_) => model) // eslint-disable-line @typescript-eslint/no-unused-vars
+}
+
+function verifyObjectDefinition(definitions: Record<string, any>, key: string): boolean {
```

Comment:
It is not clear what this function does. Please add some comments why we need it and what it does.